hrms/payroll/doctype/salary_slip/salary_slip.js

```diff
@@ -97,6 +97,8 @@ frappe.ui.form.on("Salary Slip", {
 			if (frm.doc.currency) {
 				var from_currency = frm.doc.currency;
 				if (from_currency != company_currency) {
+					if (frm.doc.exchange_rate && frm.doc.exchange_rate !==1){ return; }
```

Comment:
This seems to be an incorrect fix. If I change my currency, it will not reset my exchange rate because it will consider it to be already set and return from here.

Also in the problem you mentioned, slips are created from the payroll entry so changes to client-side code for salary slips won't affect this. The server side might not be mapping the correct exchange rate.

Reply:
So it maps values exchange rate on server side, but when we open salary slip on refresh JS triggers this method and update exchange rate value

Reply:
In that case, `set_exchange_rate` function in `salary_slip.js` should first check if the slip has a payroll entry linked. 
- If it does, the exchange rate should be fetched from the payroll entry.
- Else, it should be fetched from the standard `erpnext.setup.utils.get_exchange_rate` method